### ISSUE TITLE
PokerGPU assigns the button as the big blind in heads-up games

### ISSUE THREAD

**Issue by cerredz** [cerredz/Pulselib#14]
## Intent
`PokerGPU.reset()` assigns blinds and first action incorrectly in 2-player games. The button becomes the big blind and the non-button seat acts first preflop, which reverses heads-up poker positions and corrupts every heads-up hand from the initial deal.

## Problem Description
Current behavior:
- For `active_players == 2`, `reset()` computes `sb = button + 1` and `bb = button + 2 (mod 2)`.
- That makes the button seat the big blind, not the small blind.
- `idx` is then set to `(bb + 1) % active_players`, so the non-button seat acts first preflop.

Expected behavior:
- In heads-up poker, the button posts the small blind and acts first preflop.
- The other player posts the big blind and acts first postflop.

Conditions:
- This manifests on every 2-player hand.
- It is part of the normal training path because `trainGPU.py` resets with `active_players=True`, allowing 2-player tables.

## Affected Files
- `environments/Poker/PokerGPU.py` - computes blind seats and first actor using the same formula for all table sizes, which is wrong for heads-up.
- `scripts/Poker/trainGPU.py` - opts into variable active-player counts, so 2-player hands are expected during training.

## Code Snippets
```python
# environments/Poker/PokerGPU.py - lines 112-117
self.button = (self.button + 1) % self.active_players if hasattr(self, 'button_pos') else torch.zeros(self.n_games, dtype=torch.int32, device=self.device)
self.button_pos=self.button[0]
self.sb = (self.button + 1) % self.active_players
self.bb = (self.button + 2) % self.active_players
self.post_blinds()
self.idx = (self.bb + 1) % self.active_players
```
These formulas treat heads-up exactly like multiway play. With `active_players == 2`, `self.bb` becomes `self.button`, so the button posts the big blind and `self.idx` points to the wrong preflop actor.

```python
# scripts/Poker/trainGPU.py - lines 33-37
state, info = env.reset(options={
    'rotation': rotations,
    'active_players': True,
    'q_agent_seat': q_seat
})
```
The trainer explicitly requests variable player counts, so heads-up hands are part of the mainline environment usage.

## Root Cause Analysis
The root cause is that `environments/Poker/PokerGPU.py` uses the same seat formulas for all table sizes. Those formulas are valid for 3+ players, but heads-up poker has a special blind/order rule: the button is the small blind. The current code does not branch on `active_players == 2`, so it deterministically assigns the button as big blind and points the first preflop action at the wrong seat.

## Impact Assessment
- Affected flows: every 2-player hand in the GPU environment.
- User impact: heads-up hands are played with reversed positional rules.
- Data integrity risk: critical. Preflop action order and blind ownership are foundational game rules, so reward/policy data from heads-up hands is mislabeled from the first action.
- Failure mode: silent. The environment runs normally but models illegal poker rules.
- Severity: Critical. This is a full rule inversion on a supported table size.

## Potential Fixes
**Option A - Add a heads-up branch in `reset()`**
When `active_players == 2`, set `sb = button`, `bb = (button + 1) % 2`, and `idx = button`. This is the smallest rule-correct fix.

**Option B - Extract blind/actor assignment into a table-size-aware helper**
Create one helper that returns `(sb, bb, first_actor_preflop)` for both heads-up and multiway games. This makes the special-case explicit and testable.

**Option C - Add a dedicated heads-up regression test**
Add a test that asserts button/sb/bb/idx after `reset()` for a 2-player table. This should accompany either code fix.

## Reproduction Steps
1. Instantiate `PokerGPU` with `n_players=2`, `max_players=2`, `n_games=1`.
2. Call `env.reset(options={"active_players": False, "q_agent_seat": 0, "rotation": 0})`.
3. Inspect `button`, `sb`, `bb`, `idx`, and `current_round_bet`.
4. Actual result on the first hand:
   - `button = 0`
   - `sb = 1`
   - `bb = 0`
   - `idx = 1`
   - `current_round_bet = [1, 0]`
5. Expected result:
   - button seat `0` should be the small blind
   - seat `1` should be the big blind
   - seat `0` should act first preflop

## Additional Context
- A second reset flips `button` to `1`, but the bug persists: the button still becomes the big blind. The issue is the formula, not the initial button position.
- The CPU poker environment in `environments/Poker/Poker.py` uses the same blind formula, so the heads-up rule bug likely exists there too, but this issue is scoped to the GPU environment requested here.
